pupa/utils/generic.py

```diff
@@ -72,3 +73,20 @@ def convert_pdf(filename, type='xml'):
     data = pipe.read()
     pipe.close()
     return data
+
+
+def canonize_url(url):
+    split_url = urlsplit(url)
+
+    canonical_query = urlencode(sorted(parse_qsl(split_url.query)))
```

Comment:
So, ordering of URL params can sometimes matter - e.g. when you're using lists and stuff. Since this is always opt-in, this might be funny.

What's the usecase here? Non-deterministic ordering of params?